clippy_lints/src/returns.rs

```diff
@@ -418,19 +418,29 @@ fn check_final_expr<'tcx>(
             match cx.tcx.hir().attrs(expr.hir_id) {
                 [] => {},
                 [attr] => {
-                    if matches!(Level::from_attr(attr), Some(Level::Expect(_)))
-                        && let metas = attr.meta_item_list()
-                        && let Some(lst) = metas
-                        && let [NestedMetaItem::MetaItem(meta_item)] = lst.as_slice()
-                        && let [tool, lint_name] = meta_item.path.segments.as_slice()
-                        && tool.ident.name == sym::clippy
-                        && matches!(
-                            lint_name.ident.name.as_str(),
-                            "needless_return" | "style" | "all" | "warnings"
-                        )
-                    {
-                        // This is an expectation of the `needless_return` lint
-                    } else {
+                    if !matches!(Level::from_attr(attr), Some(Level::Expect(_))) {
+                        return;
```

Comment:
Imho do not think this approach with return for each conditions better then just list of them with &&

Reply:
The false positive was being triggered by a condition that was in the
middle of the list. Early returns made it easier to debug.

Reply:
I reverted back to a list of conditions with &&